[Navigation] Detect hasUAVisualTransition for NavigateEvent

rwlbuis opened the pull request:
#### d55a9c43bd6e450433aef6841e21b532c1c5db4b
<pre>
[Navigation] Detect hasUAVisualTransition for NavigateEvent
<a href="https://bugs.webkit.org/show_bug.cgi?id=272019">https://bugs.webkit.org/show_bug.cgi?id=272019</a>

Reviewed by Simon Fraser.

Detect hasUAVisualTransition for NavigateEvent as specified:
<a href="https://html.spec.whatwg.org/multipage/nav-history-apis.html#navigate-event-firing">https://html.spec.whatwg.org/multipage/nav-history-apis.html#navigate-event-firing</a>:dom-navigateevent-hasuavisualtransition

* LayoutTests/swipe/navigate-event-back-swipe-verify-ua-transition-expected.txt: Added.
* LayoutTests/swipe/navigate-event-back-swipe-verify-ua-transition.html: Added.
* Source/WebCore/page/Navigation.cpp:
(WebCore::Navigation::innerDispatchNavigateEvent):

Canonical link: <a href="https://commits.webkit.org/277129@main">https://commits.webkit.org/277129@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0c2ead7a057b9287d0bebd88dac39a0ed6629038

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/46233 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/25364 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/48818 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/48905 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/42274 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/48540 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/29714 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/22821 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/37774 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/46811 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/22451 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/39874 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/18986 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/19841 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/40974 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/4277 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/42546 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/41328 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/50714 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/21232 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/17742 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/44968 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/22529 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/40031 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/43882 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10336 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/22896 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/22223 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

TingPing commented on Source/WebCore/page/Navigation.cpp:
```diff
@@ -438,6 +438,8 @@ bool Navigation::innerDispatchNavigateEvent(NavigationNavigationType navigationT
     m_suppressNormalScrollRestorationDuringOngoingNavigation = false;
 
+    event->setHasUAVisualTransition(document->page() && document->page()->isInSwipeAnimation());
+
```
This would go up a few lines in `NavigateEvent::Init`

rwlbuis committed:
[Navigation] Detect hasUAVisualTransition for NavigateEvent
https://bugs.webkit.org/show_bug.cgi?id=272019

Reviewed by Simon Fraser.

Detect hasUAVisualTransition for NavigateEvent as specified:
https://html.spec.whatwg.org/multipage/nav-history-apis.html#navigate-event-firing:dom-navigateevent-hasuavisualtransition

* LayoutTests/swipe/navigate-event-back-swipe-verify-ua-transition-expected.txt: Added.
* LayoutTests/swipe/navigate-event-back-swipe-verify-ua-transition.html: Added.
* Source/WebCore/page/Navigation.cpp:
(WebCore::Navigation::innerDispatchNavigateEvent):

Canonical link: https://commits.webkit.org/277129@main
